client.go

```diff
@@ -130,19 +130,18 @@ func (c *Client) newRequestWithContext(ctx context.Context, method, pathPart str
 	return req, nil
 }
 
-// do decodes response body into v
+// do executes request and decodes response body.
 func (c *Client) do(req *http.Request, v interface{}) error {
 	resp, err := c.httpClient.Do(req)
 	if err != nil {
 		return err
 	}
 	defer resp.Body.Close()
 
-	err = json.NewDecoder(resp.Body).Decode(v)
-	return err
+	return json.NewDecoder(resp.Body).Decode(v)
```

Comment:
Can this return an error? Or does it matter it’s not handled?

Reply:
https://golang.org/pkg/encoding/json/#Decoder.Decode only returns an error